### PR TITLE
Add HTTPS related configuration for hotpot

### LINKED ISSUE #590: Hotpot lacks HTTPS related configuration
**Is this a BUG REPORT or FEATURE REQUEST?**:
> /kind feature

**What happened**:
The current hotpot lacks protection for http data transmission. For the security of data transmission, HTTPS adds SSL layer which relies on certificates to verify the identity of the server and encrypt the communication between the browser and the server.

**What you expected to happen**:
Add HTTPS related configuration for beego server and client in hotpot.

**Anything else we need to know?**:
Add certificate verification on the client side and beego server side. Add server time out and 
cryptographic kit on beego server.

**Environment**:
- Hotpot(release/branch) version: latest version

### PR DESCRIPTION
**Pull request by lasshpanzer**
**What this PR does / why we need it**:
Add HTTPS related configuration for hotpot

**Which issue this PR fixes**: fixes #590

### CODE REVIEW

File: pkg/api/router.go [hunk 1]
```diff
@@ -20,8 +20,13 @@ This module implements a entry into the OpenSDS northbound REST service.
 package api
 
 import (
+	"crypto/tls"
 	"fmt"
 	"net/http"
+	"strconv"
+	"strings"
+
+	. "github.com/opensds/opensds/pkg/utils/config"
```
Comment (leonwanghui) [inline pkg/api/router.go]: I think it's better to not treat `config` as internal package.

File: pkg/api/router.go [hunk 2]
```diff
@@ -111,6 +116,30 @@ func Run(host string) {
 	beego.Router("/", &VersionPortal{}, "get:ListVersions")
 	beego.Router("/:apiVersion", &VersionPortal{}, "get:GetVersion")
 
+	// beego https config
+	beego.BConfig.Listen.EnableHTTP = false
+	beego.BConfig.Listen.EnableHTTPS = true
+	strs := strings.Split(host.ApiEndpoint, ":")
+	beego.BConfig.Listen.HTTPSAddr = strs[0]
```
Comment (leonwanghui) [inline pkg/api/router.go]: I suggest to construct `iota` variables to represent Address and Prot

File: script/devsds/lib/certificate.sh [hunk 3]
```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2018 Huawei Technologies Co., Ltd. All Rights Reserved.
```
Comment (leonwanghui) [inline script/devsds/lib/certificate.sh]: Please change copyright time to 2019

File: pkg/api/router.go (hunk 1 above)
Comment (lasshpanzer) [reply pkg/api/router.go]: Changed.

File: pkg/api/router.go (hunk 2 above)
Comment (lasshpanzer) [reply pkg/api/router.go]: Changed.

**Pull request merged**